setuptools/command/register.py

```diff
@@ -1,10 +1,24 @@
+from distutils import log
 import distutils.command.register as orig
+from distutils.errors import DistutilsError
 
 
 class register(orig.register):
     __doc__ = orig.register.__doc__
 
+    def _deprecation_notice(self):
+        self.announce(
+            "WARNING: Registering is deprecated, use twine to "
+            "upload instead (https://pypi.org/p/twine/)",
+            log.WARN
+        )
+
     def run(self):
-        # Make sure that we are using valid current name/version info
-        self.run_command('egg_info')
-        orig.register.run(self)
+        try:
+            # Make sure that we are using valid current name/version info
+            self.run_command('egg_info')
+            orig.register.run(self)
+            self._deprecation_notice()
```

Comment:
I'm not sure I understand why the try/catch here.

Can we do this?

```python
self._deprecation_notice()
self.run_command('egg_info')
orig.register.run(self)
```
Is it because `run_command` and `register.run` generate a bunch of text, and you want the deprecation warning to be printed *after* this text?

If so, maybe this would be better:

```
try:
    self.run_command('egg_info')
    orig.register.run(self)
except Exception:
    raise
finally:
    self._deprecation_notice()
```